source/lib/src/prod_env_mat.cc

```diff
@@ -175,6 +175,111 @@ prod_env_mat_r_cpu(
   }
 }
 
+template<typename FPTYPE>
+void
+deepmd::
+prod_env_mat_a_mix_cpu(
+    FPTYPE * em, 
+    FPTYPE * em_deriv, 
+    FPTYPE * rij, 
+    int * nlist, 
+    const FPTYPE * coord, 
+    const int * f_type, 
+    const int * type, 
+    const InputNlist & inlist,
+    const int max_nbor_size,
+    const FPTYPE * avg, 
+    const FPTYPE * std, 
+    const int nloc, 
+    const int nall, 
+    const float rcut, 
+    const float rcut_smth, 
+    const std::vector<int> sec) 
+{
+  const int nnei = sec.back();
+  const int nem = nnei * 4;
+  memset(em, 0.0, sizeof(FPTYPE) * nloc * nem);
+  memset(em_deriv, 0.0, sizeof(FPTYPE) * nloc * nem * 3);
+  memset(rij, 0.0, sizeof(FPTYPE) * nloc * nnei * 3);
```

Comment:
The second parameter of `memset` is passed as an `int`. See https://cplusplus.com/reference/cstring/memset/